### PR TITLE
Fix API with vite proxy

### PR DESCRIPTION
**Pull request by xcodebuild**
fix https://github.com/OpenDevin/OpenDevin/issues/441

`VITE_WS_URL` and `VITE_API_URL` is not necessary now

- `http://localhost:3001/api`(frontend) -> `http://localhost:3000/`(backend)
- `ws://localhost:3001/ws`(frontend) -> `ws://localhost:3000/ws`(backend)

### CODE REVIEW

**Comment by yimothysu** [comment]
@xcodebuild can you explain the rationale behind this change? I'm hesitant about putting these settings in `vite.config.ts`.

I prefer environment variables because they allow for greater flexibility. If we were to deploy this for example, we might have a different frontend and backend hostname, and it would be more streamlined to switch the URL environment variables than set a new `vite.config.ts`.

**Comment by xcodebuild** [comment]
Vite provides a proxy to forward front-end requests to the back-end application, so the front-end application doesn't need to worry about where the back-end application is, it just needs to request `/api/xxx`.

Environment variables are fine, but at least we shouldn't be using a lot of environment variables for one thing. Maybe providing a `BACKEND_HOST_URL` or `BACKEND_HOST_PORT` and then letting `vite.config.ts` and the back-end application to consume it together would be good.

**Comment by yimothysu** [comment]
I see. Can we define a single environment variable `BACKEND_HOST_URL`? We can use this variable in `vite.config.ts`.

**Comment by xcodebuild** [comment]
`BACKEND_HOST` and related checks have already been added.

File: frontend/src/socket/socket.ts [hunk 1]
```diff
@@ -6,12 +6,7 @@ import { handleObservationMessage } from "./observations";
 
 type SocketMessage = ActionMessage | ObservationMessage;
 
-const WS_URL = import.meta.env.VITE_WS_URL;
-if (!WS_URL) {
-  throw new Error(
-    "The environment variable VITE_WS_URL is not set. Please set it to the WebSocket URL of the terminal server.",
-  );
-}
+const WS_URL = `ws://${window.location.host}/ws`;
```
Comment (yimothysu) [inline frontend/src/socket/socket.ts:9]: Can we replace with `const WS_URL = "/ws";`? Will this proxy via vite config?

Comment (xcodebuild) [reply frontend/src/socket/socket.ts:9]: No, `/ws/` is not a valid websocket URL. But `ws://${window.location.host}/ws` will be proxy via vite.

Comment (yimothysu) [reply frontend/src/socket/socket.ts:9]: Got it.

**Review by yimothysu — APPROVED** [review]

**Pull request merged**